pkg/osquery/tables/airport/table_darwin.go

```diff
@@ -0,0 +1,262 @@
+			return nil, err
+		}
+		results = append(results, optionResult...)
+	}
+
+	return results, nil
+}
+
+func processAirportOutput(airportOutput io.Reader, option string, queryContext table.QueryContext, logger log.Logger) ([]map[string]string, error) {
+	var results []map[string]string
+
+	var unmarshalledOutput []map[string]interface{}
+
+	rowData := map[string]string{"option": option}
+
+	switch option {
+	case "getinfo":
+		unmarshalledOutput = []map[string]interface{}{unmarshallGetInfoOutput(airportOutput)}
+	case "scan":
+		unmarshalledOutput = unmarshallScanOuput(airportOutput)
+	default:
+		return nil, errors.Errorf("unsupported option %s", option)
+	}
+
+	for _, dataQuery := range tablehelpers.GetConstraints(queryContext, "query", tablehelpers.WithDefaults("*")) {
+
+		flattened, err := dataflatten.Flatten(unmarshalledOutput, dataflatten.WithLogger(logger), dataflatten.WithQuery(strings.Split(dataQuery, "/")))
+		if err != nil {
+			return nil, err
+		}
+		results = append(results, dataflattentable.ToMap(flattened, dataQuery, rowData)...)
+	}
+
+	return results, nil
+}
+
+// unmarshallScanOuput parses the output of the airport getinfo command
+func unmarshallGetInfoOutput(reader io.Reader) map[string]interface{} {
+	/* example output:
+
+	    agrCtlRSSI: -55
+	    agrExtRSSI: 0
+	   agrCtlNoise: -94
+	   agrExtNoise: 0
+	         state: running
+	       op mode: station
+	       ...
+	*/
+	scanner := bufio.NewScanner(reader)
+	scanner.Split(bufio.ScanLines)
+
+	results := make(map[string]interface{})
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		// since BSSID (a1:a1:a1...) contains ":", we need to split on ": "
+		parts := trimSpaces(strings.Split(line, ": "))
+
+		// if we only have a key, move on
+		if len(parts) < 2 {
+			continue
+		}
+
+		results[parts[0]] = parts[1]
+	}
+
+	return results
+}
+
+// unmarshallScanOuput parses the output of the airport scan command
+func unmarshallScanOuput(reader io.Reader) []map[string]interface{} {
+	/* example output:
+
+	            SSID BSSID             RSSI CHANNEL HT CC SECURITY (auth/unicast/group)
+	   i got spaces! a0:a0:a0:a0:a0:a0 -92  108     Y  US WPA(PSK/AES,TKIP/TKIP) RSN(PSK/AES,TKIP/TKIP)
+	       no-spaces b1:b1:b1:b1:b1:b1 -91  116     N  EU RSN(PSK/AES/AES)
+	*/
+
+	scanner := bufio.NewScanner(reader)
+	scanner.Split(bufio.ScanLines)
+
+	var headers []string
+	var headerSeparatorIndexes []int
+	var results []map[string]interface{}
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if headerSeparatorIndexes == nil {
+			headerSeparatorIndexes = columnSeparatorIndexes(line)
+
+			if len(headerSeparatorIndexes) > 1 {
+				headerSeparatorIndexes = headerSeparatorIndexes[:len(headerSeparatorIndexes)-1]
+			}
+
+			headers = splitAtIndexes(line, headerSeparatorIndexes)
+
+			continue
+		}
+
+		rowData := make(map[string]interface{}, len(headers))
+
+		for i, value := range splitAtIndexes(line, headerSeparatorIndexes) {
+
+			if i >= len(headers) {
+				break
```

Comment:
actually, now that I have stared at it for a while, I don't think it's possible to ever reach this condition since we are splitting based on the size of the header indexes ... I actually added a condition to the splitAtIndexes function to make sure the reverse doesn't cause a panic (where the line is to short to be split by column separator indexes)